### PR TITLE
Add characterization tests for pure-logic classes

### PR DESCRIPTION
**Pull request by jasonschroeder-sfdc**
## Summary

Raises the test-coverage floor by adding characterization tests for four pure-logic classes that had low or zero coverage. **No production code is changed.** Tests use only in-memory fixtures (Mockito where needed) — no Redis, Docker, or HTTP dependencies.

| Class | Line coverage before → after |
|---|---|
| `instance/stub/Chunker` | 33% → 82% |
| `common/TreeIterator` | 6% → 92% |
| `worker/PutOperationStage` | 0% → 95% |
| `worker/resources/LocalResourceSetUtils` | 43% → 78% |

## Details

- **Chunker** — builder variants, the (intentionally inverted) `hasNext()`/`next()` iteration contract, empty-input handling, chunk splitting, `seek`/`reset`, `Chunk` equals/hashCode, EOF.
- **TreeIterator** — DFS traversal, empty/missing directories, pagination token round-trip, malformed token. Backed by an in-memory `DirectoryFetcher`.
- **PutOperationStage** — `put()` forwarding, interrupt propagation, `hasResult` vs partial-metadata branches, period configuration. (Wall-clock-dependent averaging is deliberately not asserted on value.)
- **LocalResourceSetUtils** — `create` (semaphore + pool), `exhausted`, `claimResources` (pool acquire, insufficient semaphore, partial-claim rollback, non-numeric value → 1 unit), `getResourceName`, `satisfies`.

The only non-test change is adding a direct `remoteapis` proto dependency to the `common` test `BUILD`, required to compile `TreeIteratorTest`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/test/java/build/buildfarm/common/TreeIteratorTest.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
+// Copyright 2024 The Buildfarm Authors. All rights reserved.
```
Comment (werkt) [inline src/test/java/build/buildfarm/common/TreeIteratorTest.java:1]: Copyright year

File: src/test/java/build/buildfarm/instance/stub/ChunkerTest.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
+// Copyright 2024 The Buildfarm Authors. All rights reserved.
```
Comment (werkt) [inline src/test/java/build/buildfarm/instance/stub/ChunkerTest.java:1]: Copyright year

File: src/test/java/build/buildfarm/worker/PutOperationStageTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 The Buildfarm Authors. All rights reserved.
```
Comment (werkt) [inline src/test/java/build/buildfarm/worker/PutOperationStageTest.java:1]: Copyright year

**Review by werkt — CHANGES_REQUESTED** [review]
